flink-table-store-core/src/main/java/org/apache/flink/table/store/file/manifest/ManifestFileMeta.java

```diff
@@ -135,19 +135,22 @@ public String toString() {
             List<ManifestFileMeta> metas,
             List<ManifestEntry> entries,
             ManifestFile manifestFile,
-            long suggestedMetaSize) {
+            long suggestedMetaSize,
+            int suggestedMinMetaCount) {
         List<ManifestFileMeta> result = new ArrayList<>();
         // these are the newly created manifest files, clean them up if exception occurs
         List<ManifestFileMeta> newMetas = new ArrayList<>();
         List<ManifestFileMeta> candidate = new ArrayList<>();
         long totalSize = 0;
+        int metaCount = 0;
 
         try {
             // merge existing manifests first
             for (ManifestFileMeta manifest : metas) {
                 totalSize += manifest.fileSize;
+                metaCount += 1;
                 candidate.add(manifest);
-                if (totalSize >= suggestedMetaSize) {
+                if (totalSize >= suggestedMetaSize && metaCount >= suggestedMinMetaCount) {
```

Comment:
We should choose not to merge when the totalSize is not satisfied and the min count is not satisfied.
So we need check before below `mergeIntoOneFile`.

Reply:
> 

Hey @JingsongLi , thanks for your review! Please correct me if I didn't understand it correctly, do you mean we should have this check in `mergeIntoOneFile`?

Reply:
> We should choose not to merge when the totalSize is not satisfied and the min count is not satisfied. So we need check before below `mergeIntoOneFile`.

Hey @JingsongLi thanks for your review! Please correct me if I didn't understand it correctly, do you mean we should have this check in `mergeIntoOneFile`?

Reply:
You're missing the case when we merge the last bit of manifests with entries. See line 172.